src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -431,9 +432,10 @@ private IList<Key> RegenerateKeys(IList<Key> list)
         private async Task PersistSecretsAsync<T>(T secrets, string keyScope = null, bool isNonDecryptable = false) where T : ScriptSecrets
         {
             ScriptSecretsType secretsType = secrets.SecretsType;
-            string secretsContent = ScriptSecretSerializer.SerializeSecrets<T>(secrets);
             if (isNonDecryptable)
             {
+                secrets.DecryptionKey = SystemEnvironment.Instance.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey);
```

Comment:
We don't want to persist the actual key, only a key hash... this information should be kept private on the site.